### PR TITLE
Rebase spiceai onto main branch

### PR DESCRIPTION
**Pull request by hozan23**

### CODE REVIEW

**Review by phillipleblanc — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -42,7 +41,7 @@ mysql_async = { version = "0.34.1", features = ["native-tls-tls", "chrono"], opt
 prost = { version = "0.12" , optional = true } # pinned for arrow-flight compat
 r2d2 = { version = "0.8.10", optional = true }
 rusqlite = { version = "0.31.0", optional = true }
-sea-query = { version = "0.31.0", features = ["backend-sqlite", "backend-postgres", "postgres-array", "with-rust_decimal", "with-bigdecimal", "with-time", "with-chrono"] }
+sea-query = { git = "https://github.com/spiceai/sea-query.git", rev = "213b6b876068f58159ebdd5852604a021afaebf9", features = ["backend-sqlite", "backend-postgres", "postgres-array", "with-rust_decimal", "with-bigdecimal", "with-time", "with-chrono"] }
```
Comment (phillipleblanc) [inline Cargo.toml:44]: This will prevent me from publishing a new version of this crate

File: Cargo.toml [hunk 2]
```diff
@@ -99,3 +99,7 @@ flight = [
 ]
 duckdb-federation = ["duckdb"]
 sqlite-federation = ["sqlite"]
+postgres-federation = ["postgres"]
+
+[patch.crates-io]
+duckdb = { git = "https://github.com/spiceai/duckdb-rs.git", rev = "f2ca47d094a5636df8b9f3792b2f474a7b210dc1" }
```
Comment (phillipleblanc) [inline Cargo.toml:105]: Same here

File: Cargo.toml (hunk 1 above)
Comment (hozan23) [reply Cargo.toml:44]: I will try to replace it with crate version

File: Cargo.toml (hunk 2 above)
Comment (hozan23) [reply Cargo.toml:116]: I was not able to replace this with the crate version. 

We should keep this PR open, and upstream the changes from spiceai/duckdb-rs to duckdb. what do you think?

The issue is coming from this line: https://github.com/hozan23/datafusion-table-providers/blob/6e33170f28076457bbd643e4b9fc07ca2cd450ad/src/sql/db_connection_pool/dbconnection/duckdbconn.rs#L285-L287

Comment (phillipleblanc) [reply Cargo.toml:116]: Yeah, we've had the PR open for a while now. https://github.com/duckdb/duckdb-rs/pull/373

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**